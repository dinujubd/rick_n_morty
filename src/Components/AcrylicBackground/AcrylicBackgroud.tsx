import * as React from 'react';
import { Box, createStyles, makeStyles, Theme } from '@material-ui/core';
import { connect } from 'react-redux';
import { ApplicationState } from '../../Store/store';
import { CharacterChunk } from '../../Store/characters';

interface StateProps {
    currentCharacter?: CharacterChunk;
}

const useStyles = (img: string) => makeStyles((theme: Theme) =>
    createStyles({
        root: {
            left: 0,
            right: 0,
            top: 0,
            bottom: 0,
            background: `url("${img}") no-repeat center center fixed`,
            backgroundSize: 'cover',
            filter: 'blur(70px)',
            zIndex: -2,
            position: 'absolute'
        }
    }),
);

export const AcrylicBackgroud: React.FC<StateProps> = (props) => {
    const classes = useStyles(props.currentCharacter?.image ?? '')();
    return (
        <Box>
            <Box className={classes.root}></Box>
        </Box>
    );

}

const mapStateToProps = (state: ApplicationState): StateProps => ({
    currentCharacter: state.characters.currentCharacter
})

AcrylicBackgroud.displayName = 'AcrylicBackgroud'

export default connect(mapStateToProps)(AcrylicBackgroud);