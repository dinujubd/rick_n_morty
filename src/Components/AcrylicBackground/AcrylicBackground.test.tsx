import * as React from 'react';
import renderer from 'react-test-renderer';
import { AcrylicBackgroud } from './AcrylicBackgroud';

describe('<AcrylicBackgroud />', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(<AcrylicBackgroud currentCharacter={{id: 1, name: "test", image: "http://test.image.url"}} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});