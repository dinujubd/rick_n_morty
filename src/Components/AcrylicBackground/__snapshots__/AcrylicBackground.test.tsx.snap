// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AcrylicBackgroud /> should renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="MuiBox-root MuiBox-root-3 makeStyles-root-1"
  />
</div>
`;
